FirebaseAdmin/FirebaseAdmin.Tests/Auth/Providers/OidcProviderConfigTest.cs

```diff
@@ -0,0 +1,331 @@
+                RetryOptions = RetryOptions.NoBackOff,
+            });
+            var args = FirebaseAuth.Args.CreateDefault();
+            args.ProviderConfigManager = new Lazy<ProviderConfigManager>(providerConfigManager);
+            return new FirebaseAuth(args);
+        }
+
+        private void AssertOidcProviderConfig(OidcProviderConfig provider)
+        {
+            Assert.Equal("oidc.provider", provider.ProviderId);
+            Assert.Equal("oidcProviderName", provider.DisplayName);
+            Assert.True(provider.Enabled);
+            Assert.Equal("CLIENT_ID", provider.ClientId);
+            Assert.Equal("https://oidc.com/issuer", provider.Issuer);
+        }
+
+        private void AssertClientVersionHeader(MockMessageHandler.IncomingRequest request)
+        {
+            Assert.Contains(ClientVersion, request.Headers.GetValues("X-Client-Version"));
+        }
+
+        public class InvalidCreateArgs : IEnumerable<object[]>
+        {
+            public IEnumerator<object[]> GetEnumerator()
+            {
+                // { InvalidInput, ExpectedError }
+                yield return new object[]
+                {
+                    new OidcProviderConfigArgs(),
+                    "OIDC provider ID must have the prefix 'oidc.'.",
+                };
+                yield return new object[]
+                {
+                    new OidcProviderConfigArgs()
+                    {
+                        ProviderId = string.Empty,
+                    },
+                    "OIDC provider ID must have the prefix 'oidc.'.",
+                };
+                yield return new object[]
+                {
+                    new OidcProviderConfigArgs()
+                    {
+                        ProviderId = "saml.provider",
+                    },
+                    "OIDC provider ID must have the prefix 'oidc.'.",
+                };
+                yield return new object[]
+                {
+                    new OidcProviderConfigArgs()
+                    {
+                        ProviderId = "oidc.provider",
+                    },
+                    "Client ID must not be null or empty.",
+                };
+                yield return new object[]
+                {
+                    new OidcProviderConfigArgs()
+                    {
+                        ProviderId = "oidc.provider",
+                        ClientId = "CLIENT_ID",
+                    },
+                    "Issuer must not be null or empty.",
+                };
+                yield return new object[]
+                {
+                    new OidcProviderConfigArgs()
+                    {
+                        ProviderId = "oidc.provider",
+                        ClientId = "CLIENT_ID",
+                        Issuer = "not a url",
+                    },
+                    "Malformed issuer string: not a url",
```

Comment:
There might be a missing `.` at the end

Reply:
Since we are appending user input to a string here, I think it's best to not include any additional punctuation.